src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php

```diff
@@ -0,0 +1,152 @@
+    {
+        return new RuleDefinition(
+            'Replaces ArgumentValueResolverInterface by ValueResolverInterface',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+
+final class EntityValueResolver implements ArgumentValueResolverInterface
+{
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+
+final class EntityValueResolver implements ValueResolverInterface
+{
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    private function shouldRefactorClass(Class_ $class): bool
+    {
+        // Check if the class implements ArgumentValueResolverInterface
+        foreach ($class->implements as $key => $interface) {
+            if ($interface->toString() === ArgumentValueResolverInterface::class) {
+                $class->implements[$key] = new FullyQualified(ValueResolverInterface::class);
+                return true;
+            }
+        }
+
+        // If it doesn't implement ArgumentValueResolverInterface, skip
+        return false;
+    }
+
+    /**
+     * @param Class_ $node
+     * @param int $key
+     * @param ClassMethod $classMethod
+     *
+     * @return array{bool, Expr|null, Expr|null}
+     */
+    private function extractSupportsArguments(Class_ $node, int $key, ClassMethod $classMethod): array
+    {
+        $isIdentical = true;
+        $supportFirstArg = $supportSecondArg = null;
+
+        assert(is_array($classMethod->getStmts()));
+        foreach ($classMethod->getStmts() as $statement) {
+            if (!$statement instanceof Return_) {
+                continue;
+            }
+            $expression = $statement->expr;
+            if (!$expression instanceof BinaryOp) {
+                continue;
+            }
+            if ($expression instanceof NotIdentical) {
+                $isIdentical = false;
+            }
+            $supportFirstArg = $expression->left;
+            $supportSecondArg = $expression->right;
+            unset($node->stmts[$key]);
+            break; // We only need the first matching condition
+        }
+
+        return [$isIdentical, $supportFirstArg, $supportSecondArg];
+    }
+
+    private function processResolveMethod(mixed $classMethod, mixed $isIdentical, mixed $supportFirstArg, mixed $supportSecondArg): void
+    {
+        $ifCondition = $isIdentical ? new NotIdentical($supportFirstArg, $supportSecondArg)
+            : new Identical($supportFirstArg, $supportSecondArg);
+
+        assert(is_array($classMethod->getStmts()));
```

Comment:
here as well, if already checked, passs `$clasMethod->stmts` to `processResolveMethod()` method as parameter can reduce assert check.